src/rmp/src/Restructure.cpp

```diff
@@ -66,6 +71,37 @@ using utl::RMP;
 
 namespace rmp {
 
+////////////////////////////////////////////////////////////////////////
+///                        DECLARATIONS                              ///
+////////////////////////////////////////////////////////////////////////
+
+#if defined(ABC_NAMESPACE)
```

Comment:
You should define ABC_NAMESPACE in your CAKEMAKE file. That way you won't need to forward declare the modules.

Reply:
It looks like you should use ABC_USE_NAMESPACE

Reply:
resolved by including correct header abcapis.h and ignoring the abc demo.c